src/bespokelabs/curator/types/generic_batch.py

```diff
@@ -0,0 +1,22 @@
+import datetime
+from typing import Literal, Optional
+from pydantic import BaseModel
+
+
+class GenericBatchRequestCounts(BaseModel):
+    total: int
+    failed: int
+    succeeded: int
+    raw_request_counts_object: dict
+
+
+class GenericBatch(BaseModel):
+    request_file: str
+    id: str
+    created_at: datetime.datetime
+    finished_at: Optional[datetime.datetime]
+    status: Literal["submitted", "finished", "downloaded"]
+    api_key_suffix: str
+    request_counts: GenericBatchRequestCounts
+    raw_status: str
+    raw_batch: dict
```

Comment:
nit: do these need to be BaseModels are Dataclass would work?

Reply:
We are serializing them to write to a file and the loading back in later, so using Pydantic is helpful